core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -370,7 +370,7 @@ public void testRetainLastWithExpireById() {
     }
 
     // Retain last 3 snapshots, but explicitly remove the first snapshot
-    removeSnapshots(table).expireSnapshotId(firstSnapshotId).retainLast(3).commit();
+    table.expireSnapshots().expireSnapshotId(firstSnapshotId).retainLast(3).commit();
```

Comment:
So these are cases where we used to be able to do incremental cleanup before but now will fail. I'm good with that since the default now is more safe even though from an API perspective it may be less performant for specific cases. And to address the performant issue, practically at larger scales there will be some distributed procedure running and the Spark procedure does the full reachability analysis itself anyways.

Reply:
Thank you very much for this explanation. Yes, this change chooses a more safer strategy when expiring snapshots through specifying snapshot IDs, so users will not fall into the dangerous state described above.